test/e2e/util/requestserving.go

```diff
@@ -0,0 +1,114 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
+	hyperapi "github.com/openshift/hypershift/support/api"
+	supportutil "github.com/openshift/hypershift/support/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func SetupRequestServingNodePools(ctx context.Context, t *testing.T, kubeconfigPath, mgmtHCNamespace, mgmtHCName string) []*hyperv1.NodePool {
+	g := NewWithT(t)
+	mgmtParentClient := kubeClient(t, kubeconfigPath)
+	nodePoolList := &hyperv1.NodePoolList{}
+	err := mgmtParentClient.List(ctx, nodePoolList, client.InNamespace(mgmtHCNamespace))
+	g.Expect(err).ToNot(HaveOccurred(), "cannot list nodepools in management parent cluster namespace "+mgmtHCNamespace)
+
+	// filter  management cluster nodepools
+	var mgmtNodePools []hyperv1.NodePool
+	for _, nodePool := range nodePoolList.Items {
+		if nodePool.Spec.ClusterName == mgmtHCName && nodePool.Spec.Platform.AWS != nil {
+			mgmtNodePools = append(mgmtNodePools, nodePool)
+		}
+	}
+	g.Expect(len(mgmtNodePools) >= 2).To(BeTrue(), "we need at least 2 AWS management cluster nodepools in different zones")
+
+	var nodePoolA, nodePoolB *hyperv1.NodePool
+	nodePoolA = &mgmtNodePools[0]
+	for i := range mgmtNodePools[1:] {
+		nodePool := &mgmtNodePools[1+i]
+		if nodePool.Spec.Platform.AWS.Subnet.ID != nodePoolA.Spec.Platform.AWS.Subnet.ID {
+			nodePoolB = nodePool
+			break
+		}
+	}
+	g.Expect(nodePoolB).ToNot(BeNil(), "did not find 2 nodepools with different subnets in parent")
+
+	// Prepare and create nodepools for request serving components
+	reqServingNodePoolA := nodePoolA.DeepCopy()
+	reqServingNodePoolB := nodePoolB.DeepCopy()
+
+	prepareNodePool := func(np *hyperv1.NodePool) {
+		np.ObjectMeta = metav1.ObjectMeta{
+			Name:      SimpleNameGenerator.GenerateName(fmt.Sprintf("%s-reqserving-", mgmtHCName)),
+			Namespace: mgmtHCNamespace,
+		}
+		np.Status = hyperv1.NodePoolStatus{}
+		np.Spec.Replicas = pointer.Int32(1)
+		np.Spec.AutoScaling = nil
+		np.Spec.NodeLabels = map[string]string{
+			hyperv1.RequestServingComponentLabel: "true",
```

Comment:
can we please also set the OSDFleetManagerPairedNodesLabel in this Nodes?
See https://github.com/openshift/hypershift/blob/ef4dfe9856b075e016d755f23e75ebe9489cec44/hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go#L188.

We use that in OCM to make sure pods land in subnets allowed to talk to each other.